unitTests/magicbox/MagicBoxClearTest.ts

```diff
@@ -0,0 +1,34 @@
+import { MagicBoxInstance } from '../../src/magicbox/MagicBox';
+import { MagicBoxClear } from '../../src/magicbox/MagicBoxClear';
+import { $$ } from '../../src/Core';
+import { Grammar } from '../../src/magicbox/Grammar';
+
+export const MagicBoxClearTest = () => {
+  describe('MagicBoxClear', () => {
+    let elem: HTMLElement;
+    let magicBox: MagicBoxInstance;
+
+    beforeEach(() => {
+      elem = $$('div').el;
+      magicBox = new MagicBoxInstance(elem, new Grammar('Any', { Any: /.*/ }));
+    });
+
+    it('should create a container', () => {
+      const clear = new MagicBoxClear(magicBox);
+      expect($$(clear.element).hasClass('magic-box-clear')).toBe(true);
+    });
+
+    it('should create an icon', () => {
+      const clear = new MagicBoxClear(magicBox);
+      expect($$(clear.element).find('.magic-box-icon')).toBeDefined();
+    });
+
+    it('should clear text content on click', () => {
```

Comment:
I think it would be useful to add `when the magicBox contains some text, it ...` to the test description.